cmd/storj/cmd/ls.go

```diff
@@ -32,49 +35,69 @@ func init() {
 func list(cmd *cobra.Command, args []string) error {
 	ctx := process.Ctx(cmd)
 
-	so, err := getStorjObjects(ctx, lsCfg)
+	identity, err := lsCfg.Load()
+	if err != nil {
+		return err
+	}
+
+	bs, err := lsCfg.GetBucketStore(ctx, identity)
 	if err != nil {
 		return err
 	}
 
 	if len(args) == 0 {
-		bi, err := so.ListBuckets(ctx)
-		if err != nil {
-			return err
+		startAfter := ""
+		var items []buckets.ListItem
+		for {
+			moreItems, more, err := bs.List(ctx, startAfter, "", 0)
+			if err != nil {
+				return err
+			}
+			if len(moreItems) == 0 {
+				fmt.Println("No buckets")
+				return nil
+			}
+			items = append(items, moreItems...)
+			if !more {
+				break
+			}
+			startAfter = moreItems[len(moreItems)-1].Bucket
 		}
 
-		for _, bucket := range bi {
-			fmt.Println(bucket.Created, bucket.Name)
+		for _, bucket := range items {
+			fmt.Println(bucket.Meta.Created, bucket.Bucket)
 		}
 
 		return nil
 	}
 
-	u, err := url.Parse(args[0])
+	u, err := utils.ParseURL(args[0])
 	if err != nil {
 		return err
 	}
 
-	marker := ""
+	o, err := bs.GetObjectStore(ctx, u.Host)
```

Comment:
What if `u.Host` is empty. Shouldn't we return a user-friendly error before trying the call `bs.GetObjectStore()`?

Reply:
Done